### PR TITLE
Removes Radio MMI sybtype and adds a toggleable  radio to normal MMIs.

### PR DESCRIPTION
**Pull request by Shadowlight213**
Getting blown as a borg really sucks. Usually you're stuck there forever with your only choice being to ghost and abandon your mmi body because brain code is weird and you cant like suicide or anything.
With a radio, ad least you can call for help or listen to what's going on in the station.

New update:
Regular MMIS now have built in radios! You can manually disable an MMI radio however by using an empty MMI inhand.

### CODE REVIEW

**Comment by GunHog** [comment]
It is better if you just remove the radio MMI, and add its radio functions to the base MMI. The reason being that it is confusing that roundstart cyborgs get blown, but any ones made during are probably going to be fielding standard MMIs.

**Comment by WJohn** [comment]
I agree with gunhog, there's no reason to have both, just make radio enabled the standard (and only) option.

**Comment by borntohonk** [comment]
let it be possible to interact with the MMI before you put brains into it to disable the radio tbh fam.

**Comment by Shadowlight213** [comment]
Alrighty. I was thinking of making radio MMIs  the only MMIs but I was worried it would be too much of a buff. I'll remove the radio MMI and add a radio to the base one then. Going to make sure the radio is disabled in the syndie one though.

**Comment by GunHog** [comment]
I agree with borntohonk. It is important that antags which seek to force-borg someone are not ratted out before they can complete the insertion and relawing. **This behavior is not limited to Traitors.**

**Comment by Shadowlight213** [comment]
I know that. I was planning on adding the toggling functionality. The syndie mmi would have it be disabled by default though.

**Comment by Shadowlight213** [comment]
Ok. radio moved to default MMI

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	w_class = 3
 	origin_tech = "biotech=3"
 	var/braintype = "Cyborg"
-
+	var/obj/item/device/radio/radio = null //Let's give it a radio.
```
Comment (neersighted) [inline code/modules/mob/living/carbon/brain/MMI.dm]: We have a `radio/integrated` type for this that the cyborg mob uses... You might want to check that out.

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 2]
```diff
@@ -78,7 +84,8 @@
 
 /obj/item/device/mmi/attack_self(mob/user)
 	if(!brain)
-		user << "<span class='warning'>You upend the MMI, but there's nothing in it!</span>"
+		radio.on = !radio.on
+		user << "<span class='notice'>You toggle the MMI's radio system [radio.on==1 ? "on" : "off"].</span>"
```
Comment (neersighted) [inline code/modules/mob/living/carbon/brain/MMI.dm]: no need for ==1

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 3]
```diff
@@ -139,6 +135,9 @@
 
 	if(brainmob.stat)
 		brainmob << "<span class='warning'>Can't do that while incapacitated or dead!</span>"
+	if(!radio.on)
+		brainmob << "<span class='warning'>Your radio is disabled!</span>"
+		return
 
 	radio.listening = radio.listening==1 ? 0 : 1
```
Comment (neersighted) [inline code/modules/mob/living/carbon/brain/MMI.dm]: radio.listening = !radio.listening

**Pull request merged**